packages/slas-react-sdk/src/index.tsx

```diff
@@ -0,0 +1,235 @@
+        _config: config,
+        authType: AuthTypes.GUEST,
+        customerId: "",
+        isInitialized: false,
+        isInitializing: true,
+        isAuthenticated: false,
+        accessToken: "",
+        refreshToken: "",
+        _initialize: () => {
+          logger("INIT - start");
+          const self = get();
+          const isAuthenticated = !!(
+            self.accessToken && !isExpired(self.accessToken)
+          );
+
+          if (isAuthenticated) {
+            logger("Found access token and it's not expired, done.");
+            set({
+              isInitializing: false,
+              isInitialized: true,
+              isAuthenticated,
+            });
+            return;
+          }
+
+          if (self.refreshToken) {
+            logger("Found refresh token. Starting refresh token flow.");
+
+            self._refreshAccessToken();
+            return;
+          }
+
+          logger("No access/refresh token found. Starting guest user flow.");
+          self.loginAsGuest();
+        },
+        login: async (username, password) => {
+          logger("Logging in as registered user.");
+          const self = get();
+          const res = await helpers.loginRegisteredUserB2C(
+            shopperLoginClient,
+            { username, password },
+            {
+              redirectURI: `${getAppOrigin()}/callback`,
+              // usid:
+            }
+          );
+          self._handleShopperLoginTokenResponse(res);
+          logger("Logged in as registered user.");
+          // enhancement TODO: if it's useful, we could automatically
+          // fetch user information by calling /customers/customerId
+        },
+        register: async (username, password) => {},
+        loginAsGuest: async () => {
+          logger("Logging in as guest.");
+          const self = get();
+          const res = await helpers.loginGuestUser(shopperLoginClient, {
+            redirectURI: `${getAppOrigin()}/callback`,
+          });
+          self._handleShopperLoginTokenResponse(res);
+          logger("Logged in as guest.");
+        },
+        logout: async () => {
+          logger("Logging out.");
+          const self = get();
+          // There is a bug with the logout helper
+          // See https://github.com/SalesforceCommerceCloud/commerce-sdk-isomorphic/issues/88
+          // const response = await helpers.logout(shopperLoginClient, {
+          //   refreshToken: self.refreshToken,
+          // });
+
+          const response = await shopperLoginClient.logoutCustomer({
+            parameters: {
+              refresh_token: self.refreshToken,
+              client_id: self._config.clientId,
+              channel_id: self._config.siteId,
+            },
+            headers: {
+              Authorization: self.accessToken,
+            },
+          });
+          self._handleShopperLoginTokenResponse(response);
+          logger("Logged out.");
+          self.loginAsGuest();
+        },
```

Comment:
I mentioned this in the meeting, and John used a good word, but do we need to have these thin wrappers or façades over these functions.. could we not expose the function defined in the helpers and leave it at that?

Reply:
We also made mention about how we reflect the names of the 'login' method as there will be more than 1 login method.